pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/LightningKokkosSimulator.cpp

```diff
@@ -137,6 +137,20 @@ void LightningKokkosSimulator::PrintState() {
 }
 /// LCOV_EXCL_STOP
 
+void LightningKokkosSimulator::SetState(DataView<std::complex<double>, 1> &data,
+                                        std::vector<QubitIdType> &wires) {
+    std::vector<Kokkos::complex<double>> data_vector(data.begin(), data.end());
+    std::vector<std::size_t> wires_size_t(wires.begin(), wires.end());
+    this->device_sv->setStateVector(wires_size_t, data_vector);
```

Comment:
Does that work, because I think the signature is 
```
void setStateVector(const std::vector<ComplexT> &state,
                        const std::vector<std::size_t> &wires);
```
?

Reply:
Oh I know what happening, you calling 
```
void setStateVector(const std::vector<std::size_t> &indices,
                        const std::vector<ComplexT> &values)
```
but this is the sparse matrix-style interface. The test below is quite simple so it's passing. I think if you set the second entry of `state` instead of the first, it will fail.